api/src/entities/user/user.controller.ts

```diff
@@ -10,54 +10,80 @@ export class UserController {
         this.userService = userService;
     }
 
-    getAll = async (req: Request, res: Response, next: NextFunction) => {
-        const users = await this.userService.all();
-        res.json(users);
-        return next();
+    getAll = async (req: Request, res: Response) => {
+        try {
+            const users = await this.userService.all();
+            res.json(users);
+        }
+        catch (error) {
+            res.sendStatus(500);
+            console.error(`Failed to get users`, error);
+        }
     }
 
-    getById = async (req: Request, res: Response, next: NextFunction) => {
+    getById = async (req: Request, res: Response) => {
         const { id } = req.params;
-        const user = await this.userService.getById(id);
-        if (user) {
-            res.json(user);
-            return next();
-        } else {
-            res.sendStatus(404);
+        try {
+            const user = await this.userService.getById(id);
+            if (user) {
+                res.json(user);
+            } else {
+                res.sendStatus(404);
+            }
+        }
+        catch (error) {
+            res.sendStatus(500);
+            console.error(`Failed to get user with id '${id}'`, error);
         }
     }
 
-    getAutoSuggestUsers = async (req: Request, res: Response, next: NextFunction) => {
+    getAutoSuggestUsers = async (req: Request, res: Response) => {
         const limit = req.query.limit ? parseInt(req.query.limit.toString(), 10) : 10;
-        const users = await this.userService.getAutoSuggestUsers(req.params.loginSubstring.toString(), limit);
-        res.json(users);
-        return next();
+        const loginSubstring = req.params.loginSubstring;
+        try {
+            const users = await this.userService.getAutoSuggestUsers(loginSubstring, limit);
+            res.json(users);
+        }
+        catch (error) {
+            res.sendStatus(500);
+            console.error(`Failed to get users by login substring '${loginSubstring}'`, error);
+        }
     }
 
-    createUser = async (req: Request, res: Response, next: NextFunction) => {
-        const user: User = {
-            ...req.body,
-            id: uuid(),
-            isDeleted: false
+    createUser = async (req: Request, res: Response) => {
+        try {
+            const user: UserInterface = {
+                ...req.body,
+                id: uuid(),
```

Comment:
I would move the decision about uuid to DB level in future PRs, since `id` seems to be the responsibility of the database level